### PR TITLE
[MRG] More lenient handling of edf channel names.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3790

Now a running number is added two the duplicates (leaving the first name untouched).
Not sure how to test this.

### CODE REVIEW

File: mne/io/edf/edf.py [hunk 1]
```diff
@@ -357,7 +357,14 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, preload):
...
     tal_ch_name = 'EDF Annotations'
-    if tal_ch_name in ch_names:
+    tal_chs = np.where(np.array(ch_names) == tal_ch_name)[0]
+    if len(tal_chs) > 0:
+        if len(tal_chs) > 1:
+            warn('Channel names are not unique, found duplicates for: %s. '
+                 'Adding running numbers to duplicate channel names.'
+                 % tal_ch_name)
+            for idx, tal_ch in enumerate(tal_chs[1:]):
+                ch_names[tal_ch] = ch_names[tal_ch] + '-%s' % (idx + 1)
```
Comment (dengemann) [inline mne/io/edf/edf.py]: nice! btw. you could also pass 1 as second argument to enumerate to avoid the `(idx + 1 )` expression.

**Comment by jaeilepp** [comment]
So this basically only uses the first tal_channel to extract the events. Also, seems that the provided data does not have annotations with descriptions so the stim channel ends up being zeros. I don't know how edf data usually is and what to do with these extra tal_channels. Now they're just set to misc channels.

**Comment by jaeilepp** [comment]
Ok, seems that it's the second tal_channel that has the proper events. Maybe these tal_channels need to be merged...

**Comment by Eric89GXL** [comment]
@jaeilepp can you convert each one to a stim channel? Neuromag has multiple stim channels, STI001, STI002, ..., STI008, STI101, STI201, STI301...

**Comment by jaeilepp** [comment]
> @jaeilepp can you convert each one to a stim channel?

Yeah, that's a good idea.

File: mne/io/edf/edf.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         data_size = self._raw_extras[fi]['data_size']
         data_offset = self._raw_extras[fi]['data_offset']
         stim_channel = self._raw_extras[fi]['stim_channel']
-        tal_channel = self._raw_extras[fi]['tal_channel']
+        tal_channels = self._raw_extras[fi]['tal_channel']
```
Comment (Eric89GXL) [inline mne/io/edf/edf.py]: Not as input to `Raw` constructor, but I think this happens with `concatenate_raws`

**Comment by agramfort** [comment]
CIs are not happy

**Comment by jaeilepp** [comment]
> Neuromag has multiple stim channels, STI001, STI002, ..., STI008, STI101, STI201, STI301...

Now that I look at it, maybe there should be a channel that merges everything together after all. The annotation channels are still there if the user wants to use them individually. Isn't it so that Neuromag also has a channel that merges all together?

**Comment by agramfort** [comment]
yes neuromag fuses all the stim channels onto STI101

**Comment by jaeilepp** [comment]
I'm wondering how the fusing should be done. I first thought of bitwise or (like Neuromag does), but the annotations are mapped to running integers, so it would not work. Maybe it should just be added on top of each other. In my opinion the annotations should be mapped to exponentials of 2, but I'm not sure how to do it without breaking people's code.

**Comment by Eric89GXL** [comment]
You could just add, and if they want individual channels they could be
available at STI00X. It's not perfect but it sold be usable.

**Comment by jaeilepp** [comment]
Ok. Now the events are merged in channel 'STI 014' (through addition) and the individual tal channels are named 'EDF Annotations-<idx>'.

**Comment by Eric89GXL** [comment]
Sounds good to me but I don't really work with EDF data much. @teonbrooks do you have time to look?

**Comment by Eric89GXL** [comment]
We should probably modify a test for this to make sure it works properly. We can either read-modify-write (to tempdir) an existing EDF file, or get a file from someone to add to `testing`.

**Comment by dgwakeman** [comment]
hmmm. @jaeilepp @Eric89GXL @dengemann what about making `STI101` the synthesis channel? That is the standard on all the newer Neuromag systems and `STI 014` is the old standard. I think, it should for consistency's sake either use the current Neuromag standard, or a different number all together (e.g. `STI435` ; echo EDF | tr ABCDEFGHI 012345678). Thoughts?

**Comment by Eric89GXL** [comment]
I'd prefer using `STI101` if we were starting from scratch, but all of our other readers synthesize their synthetic channels to `STI 014` already :(

**Comment by agramfort** [comment]
@jaeilepp are we good here? should I be concerned about the little test added?

**Comment by jaeilepp** [comment]
Good from my side. Adding a test would require adding a test file with multiple tal channels.

**Comment by agramfort** [comment]
ok thx @jaeilepp

**Comment by agramfort** [comment]
@jaeilepp please just push a what's new update in master

thx

**Pull request merged**